addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechPanel.java

```diff
@@ -386,4 +396,67 @@ private JButton getOptionsButton() {
         }
         return optionsButton;
     }
+
+    private JToggleButton getLinkWithSitesTreeButton() {
+        if (linkWithSitesTreeButton == null) {
+            linkWithSitesTreeButton = new ZapToggleButton();
+            linkWithSitesTreeButton.setIcon(
+                    new ImageIcon(TechPanel.class.getResource("/resource/icon/16/earth-grey.png")));
+            linkWithSitesTreeButton.setToolTipText(
+                    Constant.messages.getString(
+                            "wappalyzer.toolbar.toggle.site.link.disabled.tooltip"));
+            linkWithSitesTreeButton.setSelectedIcon(
+                    new ImageIcon(TechPanel.class.getResource("/resource/icon/16/094.png")));
+            linkWithSitesTreeButton.setSelectedToolTipText(
+                    Constant.messages.getString(
+                            "wappalyzer.toolbar.toggle.site.link.enabled.tooltip"));
+            DisplayUtils.scaleIcon(linkWithSitesTreeButton);
+            setLinkWithSitesTreeSelection(true);
+            linkWithSitesTreeButton.addActionListener(
+                    e -> setLinkWithSitesTreeSelection(linkWithSitesTreeButton.isSelected()));
+        }
+        return linkWithSitesTreeButton;
+    }
+
+    private void setLinkWithSitesTreeSelection(boolean enabled) {
+        linkWithSitesTreeButton.setSelected(enabled);
+        JTree sitesTree = View.getSingleton().getSiteTreePanel().getTreeSite();
+        if (enabled) {
+            TreePath selectionPath = sitesTree.getSelectionPath();
+            if (selectionPath != null) {
+                setSiteSelection(selectionPath);
+            }
+            sitesTree.addTreeSelectionListener(getLinkWithSitesTreeSelectionListener());
+        } else {
+            sitesTree.removeTreeSelectionListener(getLinkWithSitesTreeSelectionListener());
+        }
+    }
+
+    private void setSiteSelection(TreePath selectionPath) {
+        SiteNode selected = (SiteNode) selectionPath.getLastPathComponent();
+        if (selected.isRoot()) {
+            return;
+        }
+        siteSelected(ExtensionWappalyzer.normalizeSite((selected).getHistoryReference().getURI()));
+    }
+
+    protected LinkWithSitesTreeSelectionListener getLinkWithSitesTreeSelectionListener() {
+        if (linkWithSitesTreeSelectionListener == null) {
+            linkWithSitesTreeSelectionListener = new LinkWithSitesTreeSelectionListener();
+        }
+        return linkWithSitesTreeSelectionListener;
+    }
+
+    private class LinkWithSitesTreeSelectionListener implements TreeSelectionListener {
+
+        @Override
+        public void valueChanged(TreeSelectionEvent e) {
+            setSiteSelection(e.getPath());
+        }
+    }
+
+    protected void unload() {
+        View.getSingleton().getSiteTreePanel().getTreeSite()
+                .removeTreeSelectionListener(getLinkWithSitesTreeSelectionListener());
```

Comment:
I'd suggest extracting a method and call from both places (even if that means not using the other local variable, it makes it clear that it's undoing the same).